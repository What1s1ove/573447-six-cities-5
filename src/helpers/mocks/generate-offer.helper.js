import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
