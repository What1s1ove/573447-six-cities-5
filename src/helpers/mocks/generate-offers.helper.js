import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
