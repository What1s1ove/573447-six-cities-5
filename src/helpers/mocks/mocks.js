export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
