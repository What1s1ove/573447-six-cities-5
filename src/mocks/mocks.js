export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
