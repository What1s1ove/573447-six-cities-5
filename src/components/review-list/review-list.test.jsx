import * as React from 'react';
import {render} from 'enzyme';
import {mockedReviews} from '~/mocks/mocks';
import ReviewList from './review-list';

describe(`ReviewList component`, () => {
  it(`should render ReviewList component`, () => {
    const component = render(<ReviewList reviews={mockedReviews} />);

    expect(component).toMatchSnapshot();
  });
});
