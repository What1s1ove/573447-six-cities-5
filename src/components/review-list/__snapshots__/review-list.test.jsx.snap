// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList component should render ReviewList component 1`] = `
<ul
  class="reviews__list"
>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Jack
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width:80%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        I stayed here for one night and it was an unpleasant experience.
      </p>
      <time
        class="reviews__time"
        datetime="2020-10-04T09:29:21.139Z"
      >
        October 2020
      </time>
    </div>
  </li>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Corey
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width:80%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
      </p>
      <time
        class="reviews__time"
        datetime="2020-10-04T09:29:21.139Z"
      >
        October 2020
      </time>
    </div>
  </li>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Isaac
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width:80%"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
      </p>
      <time
        class="reviews__time"
        datetime="2020-10-04T09:29:21.139Z"
      >
        October 2020
      </time>
    </div>
  </li>
</ul>
`;
